### PR TITLE
[Tests]postcopy.go: Remove use of RunVMAndExpectLaunchWithRunStrategy func

### PR DESCRIPTION
**Pull request by Barakmor1**
Add WithRunStrategy func to libvmi
WithRunStrategy is the correct way to choose
How vmi should behave and libvmi should include
option for RunStrategy setting.

postcopy.go: Remove use of RunVMAndExpectLaunchWithRunStrategy func
RunVMAndExpectLaunchWithRunStrategy should be deprecated as it is part
of tests_utils.go and use expensive get and update API calls

Instead of Updating VM runStrategy in postcopy.go create vm with
with the desired runStrategy.

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Comment by fossedihelm** [comment]
Conflict w/ https://github.com/kubevirt/kubevirt/pull/12023?

**Review by enp0s3 — COMMENTED** [review]
/approve
@Barakmor1 Thanks!
@fossedihelm I don't think it conflicts since the other PR is a draft. The author can update the draft following this PR.

File: pkg/libvmi/vm.go [hunk 1]
```diff
@@ -65,6 +65,12 @@ func WithRunning() VMOption {
 	}
 }
 
+func WithRunStrategy(strategy v1.VirtualMachineRunStrategy) VMOption {
+	return func(vm *v1.VirtualMachine) {
+		vm.Spec.RunStrategy = &strategy
```
Comment (fossedihelm) [inline pkg/libvmi/vm.go:70]: I'd expect here to have:
```
vm.Spec.Running = nil
```

Comment (dankenigsberg) [reply pkg/libvmi/vm.go:70]: Makes sense, but requires `vm.Spec.RunStrategy =  nil` after line 64 for symmetry.

File: tests/migration/postcopy.go [hunk 2]
```diff
@@ -257,22 +244,25 @@ var _ = SIGMigrationDescribe("VM Post Copy Live Migration", func() {
 						}, 120*time.Second, 1*time.Second).Should(Succeed(), "Virt handler should come online")
 					}
 					It("should make sure that VM restarts after failure", func() {
-
 						By("creating a large VM with RunStrategyRerunOnFailure")
-						vm := createLargeVirtualMachine()
+						vmi := libvmifact.NewFedora(
+							append(libnet.WithMasqueradeNetworking(),
```
Comment (dankenigsberg) [inline tests/migration/postcopy.go:249]: requires rebase now that `libnet.WithMasqueradeNetworking()` returns one option.

**Review by dankenigsberg — CHANGES_REQUESTED** [review]

File: tests/migration/postcopy.go (hunk 2 above)
Comment (Barakmor1) [reply tests/migration/postcopy.go]: Done :)

File: tests/migration/postcopy.go [hunk 3]
```diff
@@ -257,22 +244,25 @@ var _ = SIGMigrationDescribe("VM Post Copy Live Migration", func() {
...
-
 						By("creating a large VM with RunStrategyRerunOnFailure")
-						vm := createLargeVirtualMachine()
+						vmi := libvmifact.NewFedora(
+							libnet.WithMasqueradeNetworking(),
+							libvmi.WithResourceMemory("3Gi"),
+							libvmi.WithRng(),
+						)
+						vm := libvmi.NewVirtualMachine(vmi, libvmi.WithRunStrategy(v1.RunStrategyRerunOnFailure))
+						vm.Spec.Running = nil
```
Comment (fossedihelm) [inline tests/migration/postcopy.go:254]: This can be removed now

Comment (Barakmor1) [reply tests/migration/postcopy.go]: Done :)

**Review by fossedihelm — APPROVED** [review]
Thank you!

File: tests/migration/postcopy.go [hunk 4]
```diff
@@ -257,22 +244,24 @@ var _ = SIGMigrationDescribe("VM Post Copy Live Migration", func() {
...
 
-						By("Starting the VirtualMachine")
-						vm = tests.RunVMAndExpectLaunchWithRunStrategy(virtClient, vm, v1.RunStrategyRerunOnFailure)
-						vmi, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
-						Expect(err).ToNot(HaveOccurred())
-
-						By("Checking that the VirtualMachineInstance console has expected output")
-						Expect(console.LoginToFedora(vmi)).To(Succeed())
+						By("Waiting for the VirtualMachine to be ready")
+						vmi = libwait.WaitUntilVMIReady(vmi, console.LoginToFedora)
```
Comment (fossedihelm) [inline tests/migration/postcopy.go:264]: /hold
Failure is related
`s: "an empty namespace may not be set when a resource name is provided",`

Comment (Barakmor1) [reply tests/migration/postcopy.go:265]: Thanks Done :)

File: tests/migration/postcopy.go [hunk 5]
```diff
@@ -257,22 +244,25 @@ var _ = SIGMigrationDescribe("VM Post Copy Live Migration", func() {
...
 					}
 					It("should make sure that VM restarts after failure", func() {
-
 						By("creating a large VM with RunStrategyRerunOnFailure")
-						vm := createLargeVirtualMachine()
+						vmi := libvmifact.NewFedora(
+							libnet.WithMasqueradeNetworking(),
+							libvmi.WithResourceMemory("3Gi"),
+							libvmi.WithRng(),
+							libvmi.WithNamespace(testsuite.NamespacePrivileged),
```
Comment (dankenigsberg) [inline tests/migration/postcopy.go:252]: in a follow-up PR, please ensure that the VM runs in a normal namespace. Only the killer pod should be privileged

**Review by fossedihelm — APPROVED** [review]

**Review by dankenigsberg — APPROVED** [review]

File: tests/migration/postcopy.go (hunk 5 above)
Comment (Barakmor1) [reply tests/migration/postcopy.go:252]: Let's try it :)

**Comment by Barakmor1** [comment]
@fossedihelm PTAL

**Review by fossedihelm — APPROVED** [review]

**Pull request merged**